tests/scripts/allreduce_perf/allreduce_heuristic_code_gen.py

```
@@ -0,0 +1,255 @@
+
+
+@dataclass
+class Constants:
+    # 16384
+    num_tokens_bits = 15
+    hidden_size_bits = 14
+    max_num_tokens_considered = 2**num_tokens_bits
+    max_hidden_size_considered = 2**hidden_size_bits
+    oneshot_num_tokens_threshold: int = 1
+    oneshot_hidden_size_threshold = 128
+    num_tokens_list = [2**i for i in range(num_tokens_bits)]
```

Ruff (0.14.0)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)